Migration to Tomcat 8 / Java 8.

camarasu opened the pull request:
Attention, Java source and target change from 1.7 to 1.8.
The new bcprov-jdk15on dependency may have to be manually installed. It can be downloaded at this address: https://www.bouncycastle.org/download/bcprov-jdk15on-152.jar

glatard commented:
Can you test a complete workflow in a Tomcat 8 deployment, including:
* proxy creation.
* data transfers to/from the portal
* application launch and monitoring
* Dropbox links
?

glatard commented on Portal/pom.xml:
```diff
@@ -152,8 +152,8 @@ knowledge of the CeCILL-B license and that you accept its terms.
                 <configuration>
-                    <source>1.7</source>
-                    <target>1.7</target>
+                    <source>1.8</source>
```
Source and target version should be changed in project modules too.

camarasu replied:
Done

glatard commented on Portal/vip-core/src/main/java/fr/insalyon/creatis/vip/core/server/business/proxy/ProxyClient.java:
```diff
@@ -353,12 +354,19 @@ private void printKey(PrivateKey key, PrintStream out)
         ASN1InputStream derInputStream = new ASN1InputStream(inStream);
-        DERObject keyInfo = derInputStream.readObject();
+        ASN1Primitive keyInfo = derInputStream.readObject();
+        /*
```
Remove useless comment?

camarasu replied:
Done.

camarasu commented:
Tests for proxy creation, data transfers to/from the portal and application launch/monitoring are OK.

Dropbox link fails both with the new and the deplyed version of the portal. Therefore, I opened a specific ticket for it (Maven Issue #2687)

camarasu commented:
I keep this branch named "develop" for the current pull-request. I'll create a new "feature" branch for new commits.

camarasu commented:
I changed the version number of the portal (and modified modules accordingly), but, on a second thought, we may only want to change them when the next release (1.15) is ready...

glatard commented on Portal/pom.xml:
```diff
@@ -37,30 +37,30 @@ knowledge of the CeCILL-B license and that you accept its terms.
-        <vip.gatelab-applet.version>1.13</vip.gatelab-applet.version>
+        <vip.core.version>1.15</vip.core.version>
+        <vip.portal.version>1.15</vip.portal.version>
+        <vip.docs.version>1.15</vip.docs.version>
```
Only the version of modules that were updated in the current version must be changed. For instance, if vip-docs was not updated in 1.15, it must keep its previous (1.6) version.

camarasu replied:
Actually, I changed the version of all modules that have been re-compiled with Java 8 because their pom files have changed (as well as the corresponding jars, of course). If you believe these changes should not impact version numbers, I'll set them back to their previous version.
